pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -67,7 +71,7 @@ func (s *OAuthTokenSync) SyncOauthTokenHook(ctx context.Context, identity *authn
 	// token has no expire time configured, so we don't have to refresh it
 	if token.OAuthExpiry.IsZero() {
 		// cache the token check, so we don't perform it on every request
-		s.cache.Set(identity.ID, struct{}{}, getOAuthTokenCacheTTL(token.OAuthExpiry))
+		s.cache.Set(identity.ID, struct{}{}, getOAuthTokenCacheTTL(token.OAuthExpiry, time.Time{}))
```

Comment:
Based on my [other comment](), I don't think it happens often, but there is a possibility that an ID token is provided in conjunction with an Access token that never expires and just to be safe we should consider the ID token expiry here as well.